catalog-rest-service/src/main/java/org/openmetadata/catalog/events/ChangeEventHandler.java

```diff
@@ -150,6 +170,95 @@ private static ChangeEvent copyChangeEvent(ChangeEvent changeEvent) {
         .withCurrentVersion(changeEvent.getCurrentVersion());
   }
 
+  private enum CHANGE_TYPE {
+    UPDATE,
+    ADD,
+    DELETE
+  }
+
+  private List<Thread> getThreads(ContainerResponseContext responseContext) {
+    Object entity = responseContext.getEntity();
+    if (entity == null) {
+      return null; // Response has no entity to produce change event from
+    }
+
+    var entityInterface = Entity.getEntityInterface(entity);
+
+    List<FieldChange> fieldsUpdated = entityInterface.getChangeDescription().getFieldsUpdated();
```

Comment:
check this code https://github.com/open-metadata/OpenMetadata/blob/main/catalog-rest-service/src/main/java/org/openmetadata/catalog/slack/SlackWebhookEventPublisher.java#L75 . It can be cleaned and turn into ChangeEventParser that can be used for threads and for Slack notifications etc..

Reply:
Sure, that is something we should definitely take care of. I will track it via #2834